File: Julia/Serial/word_count/top5each.jl
# Read in file of helper functions
# This file incluse: cleantext, countwords
include("../../word_count_helpers.jl")

# Load the file names
dataLoc = "../../../data/word_count/";
fnames = dataLoc.*readdir(dataLoc)

# Iterate through file names
for fname in fnames
    
    # Read in file and clean the text
    f = open(fname)
    text = readlines(f)
    text = cleantext(text)

    # Count number of times each word appears
    wordCounts = countwords(text)

    # Sort and print the top 5 words with their counts
    rankedwords = sort(collect(wordCounts), by=x->x[2], rev=true)
    println(rankedwords[1:5,:])
    
end

File: Python/Serial/word_count/top5each.py
import os
import sys
sys.path.append('../../')
from word_count_helpers import *

# Load the file names
dataDir = "../../../data/word_count/"
fnames = os.listdir(dataDir)

# Iterate through file names
for fname in fnames:

     # Read in file and clean the text
    f = open(dataDir+fname, 'r', encoding='utf-8')
    text = cleantext(f.readlines())

    # Count number of times each word appears
    counts = countwords(text)

    # Sort and print the top 5 words with their counts
    top5 = sorted(counts, key=counts.get, reverse=True)[:5]
    str =''
    for k in top5:
        str=str + "%s: %s" % (k, counts[k]) + "; "
    print(str[:-2])
